### DISCUSSION TITLE
Let the debugging begin!

### DISCUSSION THREAD

**Discussion by rockjock32** [STELLA-Landsat/STELLA#49] [Debugging]
Having a number of issues 

- test5: Vis sensor is not detected at 0x49 on IC2 bus scan (green light goes on and off). Might be a bad unit as described in an earlier post?)
- test6: NIR_spectrum returns values, but read_device_temp says unit is not properly modified. (I checked with multimeter and junction is cut an new trace established)
- test7: not surprisingly no vis detected at 0X49
- test11: sd card not detected
- test13: no test pattern on screen, but step14 sensed touch. 

Clearly some of these are related. Any feedback welcome.

**Comment by paulmirel** [comment]
VIS: Unfortunately, I think you have a bad vis sensor unit. 

NIR: Hmm. Interesting. If it's returning values, then you are good to go. I'll add the temperature reading to the bug list. 

SD card: Since the display is not working, there's likely a problem with the display wiring, which can effect the SD card, which is on the same (SPI) bus. 

Display: Likely a wiring issue. I'll look at your photos. Multimeter in continuity mode, with power off, is useful here. Check that the wires connect where they are supposed to, and that nobody shorts to anybody else, is the general approach.

**Comment by paulmirel** [comment]
Wiring in the photograph of the display board looks correct. Be very thorough when checking for shorts on the #1 column.

**Comment by rockjock32** [comment]
So we built another screen and board and got the screen to work! However,  we are having the same issues with the VIS sensor. Nothing detected on 0x49 - what are the chances on two bad sensors? Also we noticed that the wiring diagram differs slightly from the instructions in that the VIN and GND are shown as connecting on the left of the VIS chip, but the step-by-step instructions have them connected on the right. That said the VIS chip is getting power but does not seem to be communicating with the processor.

**Comment by paulmirel** [comment]
If you have a spare microcontroller not yet assembled into an instrument, you can temporarily solder wires to the VIS sensor from the 3V, GND, SDA, SCL on the microcontroller, and run the test code to see if the unit is good on it's own. If you don't have a spare microcontroller, you can do the same thing by removing the 3V line from the microcontroller to the power bus, and then running additional wires to the sensor. If both sensors test out as bad, and it's been less than 30 days, you can contact Adafruit about getting replacements. 

So long as the VIS sensor is getting proper power, either power wiring is OK. 

Congratulations on getting the screen working!
